[WIP] Cache draft message per subscription

filipealva opened the pull request:
@RocketChat/ios

Closes #859

filipealva commented on Rocket.Chat/Controllers/Chat/ChatViewController.swift:
```diff
@@ -375,6 +383,12 @@ final class ChatViewController: SLKTextViewController {
     }
 
     // MARK: Subscription
+    
```
I got confused by this warning. For me it looks like you guys use to add a white line to separate different subtasks/context in the same scope (I like it, btw) as we can see in the lines right before the new ones I've added: 

![screen shot 2017-11-05 at 22 19 27](https://user-images.githubusercontent.com/6117280/32420869-c7d68ca6-c278-11e7-9bd1-43530e0efbc4.png)


Just let me know what I should do for this case and I'll update if it is needed.

artrmz replied:
In Xode settings (editor section) you can set to automatically trim whitespaces in new lines as well. Right now when you press "Enter" to go to next line Xcode adds leading whitespaces and the hound's warning is about those whitespaces in newlines - in other words - empty line should do not have any whitespaces, it should start from the very beginning.

rafaelks replied:
That's exactly it, thank you @artrmz!

filipealva replied:
Nice!

I'm going to fix it, thanks @artrmz!

rafaelks commented on Rocket.Chat/Controllers/Chat/ChatViewController.swift:
```diff
@@ -375,6 +383,12 @@ final class ChatViewController: SLKTextViewController {
+    func updateSubscriptionDraft(message: String) {
+        let userDefaults = UserDefaults.standard
+        userDefaults.set(message, forKey: subscription.rid)
+        userDefaults.synchronize()
```
Why do you need to call `synchronize()` here? I believe it should be handled automatically.

filipealva replied:
Actually this call is useless and shouldn't be used, my bad. I'm going to fix it too.

filipealva commented:
@rafaelks I've addressed your suggestion to have a new class to handle the caching. I tried to follow the pattern you used in `MessageTextCacheManager` but as a struct with static functions. 

Also I've added a `clearServerDraftMessages` function that runs on logout to avoid keeping data from servers that are not active anymore on the device. 

What do you think about this approach?

filipealva commented:
just closed this PR due to a messy rebase 🤔 , please see #932
